### ISSUE TITLE
Blazor Scaffolder

### ISSUE THREAD

**Issue by lachlanwgordon** [mfractor/mfractor-feedback#222]
I'm loving the new Scaffolder.  Even without the templates based on file name, this is useful enough just because you can create a folder and a file, or place a file in a folder, in one step.

How about we add some razor/blazor support?

This could follow a similar feature set to what you already offer for xaml. Interestingly for razor pages the compulsary/option files are bascially the opposite of what happens in xaml. For a new page/component you will always have a .razor but the code behind is optional, where as with xaml you always have a C# and the xaml is optional. It's also reasonably common to change your mind later and add a code behind.

I'm thinking these entries would be good:

# User enters `MyNewThing.razor`
Presented with options:
## Create a new razor component
MyNewThing.razor
```
@code {

}
```
## Create a new razor component with code behind
MyNewThing.razor would be blank

MyNewThing.razor.cs:
```
using System;
namespace {...}
{
  public partial class {MyNewThing}
  {
    public {MyNewThing}()
    {
    }
  }
}
```
## Create a new razor page
MyNewThing.razor
```
@page "/{mynewthing}" <!--the class name, lower case, remove suffix page if in file name, do not include this comment -->
@code {

}
```
## Create a new razor page with code behind
MyNewThing.razor
```
@page "/{mynewthing}" <!--the class name, lower case, remove suffix page if in file name, do not include this comment -->
```
MyNewThing.razor.cs:
```
using System;
namespace {...}
{
  public partial class {MyNewThing}
  {
    public {MyNewThing}()
    {
    }
  }
}
```

# User enters `MyNewThing.razor.cs`
Presented with options:
## Create a new razor component with code behind
Same as above
## Create a new razor page with code behind
Same as above

**Comment by matthewrdev** [comment]
Ah, this would be so cool! I'll aim to get this done over the weekend (scaffolds are easy to build) and included in the next release.

Might also be worth building out ASP.NET scaffolds + Android & iOS scaffolds... What do you think? 🤔

**Comment by matthewrdev** [comment]
This has been completed and will be available in the next MFractor release (4.2.2).

**Comment by ravero** [comment]
We can base ourselves on the file templates available in the New File dialog.

**Comment by matthewrdev** [comment]
@lachlanwgordon This will be published in MFractor 4.2.2 for both Mac and Windows. Closing this ticket.
